app/scripts/modules/core/delivery/create/CreatePipelineModal.spec.tsx

```diff
@@ -0,0 +1,192 @@
+import {PIPELINE_CONFIG_SERVICE, pipelineConfigService} from 'core/pipeline/config/services/pipelineConfig.service';
+import {Application} from 'core/application/application.model';
+import {APPLICATION_MODEL_BUILDER, ApplicationModelBuilder} from 'core/application/applicationModel.builder';
+import {IPipeline} from 'core/domain/IPipeline';
+
+describe('CreatePipelineModal', () => {
+  let $q: IQService;
+  let $scope: IScope;
+  let application: Application;
+  let initializeComponent: (configs?: Partial<IPipeline>[]) => void;
+  let component: CreatePipelineModal;
+
+  beforeEach(
+    mock.module(
+      APPLICATION_MODEL_BUILDER,
+      PIPELINE_CONFIG_SERVICE
+    )
+  );
+
+  beforeEach(mock.inject((_$q_: IQService, $rootScope: IScope, applicationModelBuilder: ApplicationModelBuilder) => {
+    $q = _$q_;
+    $scope = $rootScope.$new();
+    initializeComponent = (configs = []) => {
+      application = applicationModelBuilder.createApplication(
+        {
+          key: 'pipelineConfigs',
+          lazy: true,
+          loader: () => $q.when(null),
+          onLoad: () => $q.when(null),
+        },
+        {
+          key: 'strategyConfigs',
+          lazy: true,
+          loader: () => $q.when(null),
+          onLoad: () => $q.when(null),
+        }
+      );
+      application.pipelineConfigs.data = configs;
+
+      const props: ICreatePipelineModalProps = {
+        application,
+        show: true,
+        showCallback: (): void => null,
+        pipelineSavedCallback: (): void => null,
+      };
+
+      component = shallow(<CreatePipelineModal {...props}/>).instance() as CreatePipelineModal;
+    };
+  }));
+
+  describe('template instantiation', () => {
+    it('provides a default value when no templates exist', () => {
+      initializeComponent();
+      let template = component.state.templates[0];
+      expect(component.state.templates.length).toBe(1);
+      expect(template.name).toBe('None');
+      expect(template.application).toBe('app');
+      expect(template.triggers).toEqual([]);
+      expect(template.stages).toEqual([]);
+    });
+
+    it('includes the default value when templates exist', () => {
+      initializeComponent([{name: 'some pipeline'}]);
+      expect(component.state.templates.length).toBe(2);
+      expect(component.state.templates[0].name).toBe('None');
+      expect(component.state.templates[1].name).toBe('some pipeline');
+    });
+
+    it('initializes command with the default template', () => {
+      initializeComponent([ { name: 'some pipeline' } ]);
+      expect(component.state.templates.length).toBe(2);
+      expect(component.state.templates[0].name).toBe('None');
+      expect(component.state.templates[1].name).toBe('some pipeline');
+      expect(component.state.command.template.name).toBe('None');
+    });
+
+    it(`the component's state has a list of all config names to be used to determine if a name is unique`, () => {
```

Comment:
should be reworded so it's more of a sentence starting with `it`, maybe
`it("includes all config names in the component's state to be used to determine if a name is unique", () => {`
?